### PR TITLE
Lang switch

### PR DESCRIPTION
**Pull request by stinovlas**
This change is [Reviewable](https://reviewable.io/reviews/stinovlas/django-lang-switch/1)

### CODE REVIEW

File: django_lang_switch/templates/django_lang_switch/dropdown.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{% load i18n %}
+
+<div class="lang-switch">
+    <form action="{% url 'django_lang_switch:set_language' %}" method="post">{% csrf_token %}
```
Comment (ziima) [inline django_lang_switch/templates/django_lang_switch/dropdown.html]: `csrf_token` should be on a separate line.

File: django_lang_switch/templates/django_lang_switch/dropdown.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        <input name="next" type="hidden" value="{{ redirect_to }}">
+        <select name="language" onchange="this.form.submit()">
```
Comment (ziima) [inline django_lang_switch/templates/django_lang_switch/dropdown.html]: Inline JS!

File: django_lang_switch/templates/django_lang_switch/dropdown.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            {% get_current_language as LANGUAGE_CODE %}
+            {% get_available_languages as LANGUAGES %}
+            {% get_language_info_list for LANGUAGES as languages %}
+            {% for language in languages %}
+                <option value="{{ language.code }}"{% if language.code == LANGUAGE_CODE %} selected{% endif %}>
+                    {{ language.name_local|capfirst }} ({{ language.code }})
+                </option>
+            {% endfor %}
+        </select>
```
Comment (ziima) [inline django_lang_switch/templates/django_lang_switch/dropdown.html:19]: Wouldn't it be better to render the inputs from Django form?

File: django_lang_switch/tests/test_templatetags.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    @override_settings(
+        LANGUAGES=[('en', 'English'), ('cs', 'Czech'), ('ko', 'Korean')],
+        LANGUAGE_CODE='cs')
+    def test_simple_dropdown(self):
+        rendered = self.render_template(
+            '{% load lang_switch %}{% lang_switch_dropdown %}'
+        )
+        self.assertRegex(rendered, r'<option value="en">\s*English \(en\)\s*</option>')
+        self.assertRegex(rendered, r'<option value="cs" selected>\s*Česky \(cs\)\s*</option>')
+        self.assertRegex(rendered, r'<option value="ko">\s*한국어 \(ko\)\s*</option>')
```
Comment (ziima) [inline django_lang_switch/tests/test_templatetags.py]: I'd rather use HTML-aware assertions for HTML.

File: django_lang_switch/templates/django_lang_switch/dropdown.html [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{% load i18n %}
+
+<div class="lang-switch">
+    <form action="{% url 'django_lang_switch:set_language' %}" method="post">{% csrf_token %}
+        <input name="next" type="hidden" value="{{ redirect_to }}">
```
Comment (ziima) [inline django_lang_switch/templates/django_lang_switch/dropdown.html]: `redirect_to` doesn't seem to be set by the template tag.

**Review by ziima — CHANGES_REQUESTED** [review]

**Review by stinovlas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/stinovlas/django-lang-switch/1)* status: 0 of 16 files reviewed, 5 unresolved discussions (waiting on @ziima and @stinovlas)

___
*[django_lang_switch/templates/django_lang_switch/dropdown.html, line 5 at r1](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-LKf4hzBAHDIaX1uZYlH-r1-5:-LKfBt_641LO7vhEMcSw:b26ydho) ([raw file](https://github.com/stinovlas/django-lang-switch/blob/b9eb587eb67cb5a1131b234df7ef94d46d52256a/django_lang_switch/templates/django_lang_switch/dropdown.html#L5)):*
<details><summary><i>Previously, ziima (Vlastimil Zíma) wrote…</i></summary><blockquote>

`redirect_to` doesn't seem to be set by the template tag.
</blockquote></details>

Hmm, I copied it with the example from django docs, but as I see it, I could as well remove it. Or, perhaps, give the template tag non-mandatory keword argument `redirect_to`. What do you think?
___
*[django_lang_switch/templates/django_lang_switch/dropdown.html, line 6 at r1](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-LKf4hzBAHDIaX1uZYlH-r1-6:-LKfAuCI3pIqYtHimaeK:b-yr3l2) ([raw file](https://github.com/stinovlas/django-lang-switch/blob/b9eb587eb67cb5a1131b234df7ef94d46d52256a/django_lang_switch/templates/django_lang_switch/dropdown.html#L6)):*
<details><summary><i>Previously, ziima (Vlastimil Zíma) wrote…</i></summary><blockquote>

Inline JS!
</blockquote></details>

In this case, it's less complex than creating whole separate JavaScript file with element search within the page and so on...
___
*[django_lang_switch/templates/django_lang_switch/dropdown.html, line 15 at r1](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-LKf4hzBAHDIaX1uZYlH-r1-15:-LKfB7rt824j0gQCTd80:b-hk8rkb) ([raw file](https://github.com/stinovlas/django-lang-switch/blob/b9eb587eb67cb5a1131b234df7ef94d46d52256a/django_lang_switch/templates/django_lang_switch/dropdown.html#L15)):*
<details><summary><i>Previously, ziima (Vlastimil Zíma) wrote…</i></summary><blockquote>

Wouldn't it be better to render the inputs from Django form?
</blockquote></details>

This is basically the example from Django docs (https://docs.djangoproject.com/en/dev/topics/i18n/translation/#the-set-language-redirect-view). I think we should stick with it.

File: django_lang_switch/templates/django_lang_switch/dropdown.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+            {% for language in languages %}
+                <option value="{{ language.code }}"{% if language.code == LANGUAGE_CODE %} selected{% endif %}>
+                    {{ language.name_local|capfirst }} ({{ language.code }})
+                </option>
+            {% endfor %}
+        </select>
+    </form>
+</div>
+
+<script type="text/javascript" src="{% static "django_lang_switch/js/lang_switch.js" %}"></script>
```
Comment (ziima) [inline django_lang_switch/templates/django_lang_switch/dropdown.html:23]: `script` tags needs to be in `head`, otherwise it's still considered inline.

**Review by ziima — CHANGES_REQUESTED** [review]

**Review by ziima — CHANGES_REQUESTED** [review]
> Reviewed 10 of 16 files at r1, 5 of 7 files at r2.
> *[Reviewable](https://reviewable.io/reviews/stinovlas/django-lang-switch/1)* status: 15 of 18 files reviewed, 4 unresolved discussions (waiting on @ziima and @stinovlas)

___
*[django_lang_switch/templatetags/lang_switch.py, line 21 at r2](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-LKfhuVn9MO5BepX-FEC:-LKfhuVn9MO5BepX-FED:b6kbd11) ([raw file](https://github.com/stinovlas/django-lang-switch/blob/e8c3d635d0446b0bd19d4bbd2b57ca08f6458cb2/django_lang_switch/templatetags/lang_switch.py#L21)):*
> ```Python
>         return {}
>     else:
>         return {'redirect_to': redirect_to}
> ```

It's better to define the `redirect_to` context variable in both cases. Otherwise a `VariableDoesNotExist` warnings occurs.

**Review by ziima — CHANGES_REQUESTED** [review]
> Reviewed 1 of 16 files at r1, 1 of 7 files at r2, 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/stinovlas/django-lang-switch/1)* status: 17 of 19 files reviewed, 4 unresolved discussions (waiting on @ziima and @stinovlas)

___
*[django_lang_switch/templates/django_lang_switch/dropdown.html, line 23 at r2](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-LKf4hzBAHDIaX1uZYlH-r2-23:-LKfoBqF06Xsp7UwQOot:b-9tke24) ([raw file](https://github.com/stinovlas/django-lang-switch/blob/e8c3d635d0446b0bd19d4bbd2b57ca08f6458cb2/django_lang_switch/templates/django_lang_switch/dropdown.html#L23)):*
<details><summary><i>Previously, ziima (Vlastimil Zíma) wrote…</i></summary><blockquote>

`script` tags needs to be in `head`, otherwise it's still considered inline.
</blockquote></details>

Looks like it's not required by CSP.

**Review by stinovlas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/stinovlas/django-lang-switch/1)* status: 17 of 19 files reviewed, 4 unresolved discussions (waiting on @ziima and @stinovlas)

___
*[django_lang_switch/templates/django_lang_switch/dropdown.html, line 4 at r1](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-LKf4hzBAHDIaX1uZYlH-r1-4:-LKfrO6JEuGYwfTfHAYR:b-bw8e14) ([raw file](https://github.com/stinovlas/django-lang-switch/blob/b9eb587eb67cb5a1131b234df7ef94d46d52256a/django_lang_switch/templates/django_lang_switch/dropdown.html#L4)):*
<details><summary><i>Previously, ziima (Vlastimil Zíma) wrote…</i></summary><blockquote>

`csrf_token` should be on a separate line.
</blockquote></details>

Updated.
___
*[django_lang_switch/templates/django_lang_switch/dropdown.html, line 23 at r2](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-LKf4hzBAHDIaX1uZYlH-r2-23:-LKfqKR40dW6NaCSQYPz:b3kj9fo) ([raw file](https://github.com/stinovlas/django-lang-switch/blob/e8c3d635d0446b0bd19d4bbd2b57ca08f6458cb2/django_lang_switch/templates/django_lang_switch/dropdown.html#L23)):*
<details><summary><i>Previously, ziima (Vlastimil Zíma) wrote…</i></summary><blockquote>

Looks like it's not required by CSP.
</blockquote></details>

OK, lets leave it as it is.
___
*[django_lang_switch/templatetags/lang_switch.py, line 21 at r2](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-LKfhuVn9MO5BepX-FEC:-LKfrKmM1Ze74AykmNKR:b-bw8e14) ([raw file](https://github.com/stinovlas/django-lang-switch/blob/e8c3d635d0446b0bd19d4bbd2b57ca08f6458cb2/django_lang_switch/templatetags/lang_switch.py#L21)):*
<details><summary><i>Previously, ziima (Vlastimil Zíma) wrote…</i></summary><blockquote>

It's better to define the `redirect_to` context variable in both cases. Otherwise a `VariableDoesNotExist` warnings occurs.
</blockquote></details>

Updated.
___
*[django_lang_switch/tests/test_templatetags.py, line 23 at r1](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-LKf4hzBAHDIaX1uZYlM-r1-23:-LKfrMQX50CKqikTPdIs:b-bw8e14) ([raw file](https://github.com/stinovlas/django-lang-switch/blob/b9eb587eb67cb5a1131b234df7ef94d46d52256a/django_lang_switch/tests/test_templatetags.py#L23)):*
<details><summary><i>Previously, ziima (Vlastimil Zíma) wrote…</i></summary><blockquote>

I'd rather use HTML-aware assertions for HTML.
</blockquote></details>

Updated.

**Review by ziima — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r5.
> *[Reviewable](https://reviewable.io/reviews/stinovlas/django-lang-switch/1#-:-LKfzK1l7LTNgf95fVvC:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**